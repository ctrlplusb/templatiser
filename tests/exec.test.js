const exec = require('../src/exec')

describe('exec', () => {
  it('should produce the expected meta data', () => {
    const actual = exec({
      templatesDir: 'tests/templates',
      inputDir: 'tests/example',
      outputDir: 'tests/output',
      allowedFiles: ['.txt'],
      preprocessSource: x => `preprocessed-${x}`,
    })
    expect(actual).toMatchSnapshot()
  })
})
