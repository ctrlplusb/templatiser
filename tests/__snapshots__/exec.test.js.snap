// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exec should produce the expected meta data 1`] = `
Object {
  "config": Object {},
  "depth": 0,
  "directories": Array [
    Object {
      "config": Object {},
      "depth": 1,
      "files": Array [
        Object {
          "config": Object {},
          "depth": 1,
          "ext": ".txt",
          "name": "cape-town",
          "path": "tests/example/cities/cape-town.txt",
          "relativePath": "/cities",
          "templateName": "default",
          "type": "file",
        },
        Object {
          "config": Object {},
          "depth": 1,
          "ext": ".txt",
          "name": "london",
          "path": "tests/example/cities/london.txt",
          "relativePath": "/cities",
          "templateName": "default",
          "type": "file",
        },
      ],
      "name": "cities",
      "path": "tests/example/cities",
      "relativePath": "/cities",
      "templateName": "default",
      "type": "directory",
    },
    Object {
      "config": Object {},
      "depth": 1,
      "directories": Array [
        Object {
          "config": Object {},
          "depth": 2,
          "files": Array [
            Object {
              "config": Object {},
              "depth": 2,
              "ext": ".txt",
              "name": "apple",
              "path": "tests/example/food/fruit/apple.txt",
              "relativePath": "/food/fruit",
              "templateName": "default",
              "type": "file",
            },
            Object {
              "config": Object {},
              "depth": 2,
              "ext": ".txt",
              "name": "banana",
              "path": "tests/example/food/fruit/banana.txt",
              "relativePath": "/food/fruit",
              "templateName": "default",
              "type": "file",
            },
          ],
          "name": "fruit",
          "path": "tests/example/food/fruit",
          "relativePath": "/food/fruit",
          "templateName": "default",
          "type": "directory",
        },
      ],
      "name": "food",
      "path": "tests/example/food",
      "relativePath": "/food",
      "templateName": "default",
      "type": "directory",
    },
  ],
  "files": Array [
    Object {
      "config": Object {},
      "depth": 0,
      "ext": ".txt",
      "name": "foo",
      "path": "tests/example/foo.txt",
      "relativePath": "",
      "templateName": "default",
      "type": "file",
    },
  ],
  "templateName": "default",
  "type": "root",
}
`;
